nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/TransformOpValidation.java

```diff
@@ -2213,6 +2215,42 @@ public void testMergeAvgBp() {
 
     }
 
+    @Test
+    public void testReverseBp() {
+
+        Nd4j.getRandom().setSeed(12345);
+        SameDiff sd = SameDiff.create();
+        SDVariable input = sd.var(Nd4j.createFromArray(new double[][]{{2,7}, {3,5}, {4,5}}));
+        SDVariable out = new Reverse(sd, input,0,1).outputVariable();
+        out.markAsLoss();
+        String err = OpValidation.validate(new TestCase(sd)
+                .gradientCheck(true));
+        assertNull(err);
+
+
+    }
+
+    @Test
+    public void testUpsampling3dBp() {
+
+        Nd4j.getRandom().setSeed(12345);
+        SameDiff sd = SameDiff.create();
+
+        // NCDHW input
+        SDVariable input = sd.var(Nd4j.rand(DataType.DOUBLE, 2, 1, 5, 5, 5));
+        int scaleD = 2;
+        int scaleH = 2;
+        int scaleW = 2;
+        SDVariable out = new Upsampling3d(sd, input,true, scaleD, scaleH, scaleW).outputVariable();
+        out.markAsLoss();
+        String err = OpValidation.validate(new TestCase(sd)
+                .gradientCheck(true));
+        assertNull(err);
```

Comment:
Let's add a loop for the data format arg to test both the NCDWH and NDHWC cases.